File: 2220_Minimum_Bit_Flips_to_Convert_Number.java
class Solution {
    public int minBitFlips(int start, int goal) {
        int d = start ^ goal; 
        int count = 0; 
        while (d != 0) {
            d &= (d - 1);
            count++;
        }
        return count;  
    }
}

File: 2220_Minimum_Bit_Flips_to_Convert_Number.rs
impl Solution {
    pub fn min_bit_flips(start: i32, goal: i32) -> i32 {
        let mut d = start ^ goal;
        let mut count = 0;
        while d != 0 {
            d &= d - 1;
            count += 1;
        }
        count
    }
}
